packages/ra-core/src/controller/button/useDeleteWithConfirmController.tsx

```diff
@@ -95,7 +95,11 @@ const useDeleteWithConfirmController = <RecordType extends RaRecord = any>(
             event.stopPropagation();
             deleteOne(
                 resource,
-                { id: record.id, previousData: record },
+                {
+                    id: record.id,
+                    previousData: record,
+                    meta: mutationOptions?.meta,
```

Comment:
You should destructure mutationOptions earlier, otherwise when you pass them line 134, they may include the `meta` prop that react-query doesn't expect. I'm surprised TypeScript lets you dit it. 

So earlier, you should do as in `useEditController`:

https://github.com/marmelab/react-admin/blob/9b028c1e0c9ace57630530f6bfbfd0392f3031a1/packages/ra-core/src/controller/edit/useEditController.ts#L65